### DISCUSSION TITLE
Unreal Engine 5

### DISCUSSION THREAD

**Discussion by xxEoD2242** [CodexLabsLLC/Colosseum#14] [Announcements]
Hey everyone,

Due to the great work by @alonfaraj Unreal Engine 5 support is available for AirSim. So far, I have tested the main Python APIs and everything seems to be working as expected.

To get started, there is one major issue to solve:
- You need to go Visual Studio 2022 and download the .Net framework 3.0.1 (I don't know why this wasn't installed or why UE5 needs it).

You can then regenerate your project files, using the solution file to build the game.

Currently, I am seeing that you can't run this in Linux.  I think a new Blocks environment is warranted in UE5 as a refresh.

For now, we will keep the UE5 branch until all of the bugs are worked out and updated documentation is available.

**Comment by DroneMeOut** [comment]
.Net 3.0.1 isn't officially supported anymore and in the Visual Studio installer, only 3.1 is available. I've downloaded the SDK .Net 3.0.101 directly from MS and it is showing in the Add or Remove Programs utility in Windows, but I can't confirm that it is available to Visual Studio 2022 Community Edition I have installed. Do you know how I can confirm?

Thank you

**Comment by Snoeprol** [comment]
Does it work for UE5.3.2 as well? I built AirSim successfully with Ubuntu 22.04 but I'm unable to open the `Blocks` project in UE.

**Comment by robin271828** [comment]
I recommend using Cosys Airsim, they have [UE 5.5](https://github.com/Cosys-Lab/Cosys-AirSim/releases/tag/5.5-v3.3) support
